### ISSUE TITLE
Differentiate between "calories" and "stomach-filling"

### ISSUE THREAD

**Issue by Lamandus** [CleverRaven/Cataclysm-DDA#25586]
it would be an idea to actually differentiate between calories and "filling". So you actually get hungry, you eat and the hungermeter is determined by "stomach filling". So you could either build up some fat, to lower the chance of malnutrition (but slows you down/lowers your stamina) by eating less filling but high calorie food, or diet it down (filling but low on calories (anyone in for a salad?)).

We even can go an easy step further and say all food-types have the same filling (with different portions sizes [like alcohols]) so every "grab" of Junk food gives you... 20% filling. Every normal meal 50% and so on.

**Comment by nexusmrsep** [comment]
There is "stomach size" mechanic already coded recently, as "hunger" meter has been separated from "malnourishment" meter, but is based on nutrition (now calories). You can stuff your stomach up and feel full, but that would not magically fix your malnourished state (if you have it in the beginning), and it needs time and regular meals to disappear. In future portions' actual size and/or volume that is available in the code might be woven into it, but for now the simulation is not bad in that department. Reference: #24210

**Comment by Regularitee** [comment]
For anyone interested, there is a similar survival roguelike available for free called "UnReal World" (free on author's website, but costs money on steam) that's set in the ancient Scandinavian wilderness. It has a great example of such a system distinguishing nourishment and stomach filling foods, so might be worth a gander if anyone is thinking about developing something like that for CDDA.

**Comment by communistkiro** [comment]
Don't want to open another issue, but just noticed with the latest build that tainted fat correctly gives you 9 Cal, this should be per gram, not 0.19kg.

**Comment by Endovior** [comment]
Given the already-existing malnutrition and vomiting systems, I don't see any reason for the game to do any more tracking. This may lead to a few weird edge cases where a player who's out of real food can stuff themselves with a silly amount of candy to refill their nutrition meter, but there's already a game-mechanical reason to not do that (health). The alternative would seem to do little but add the possibility of stuffing yourself with junk to the point of no longer being able to eat while still being hungry, and that just feels like unnecessary tedium.

**Comment by Lamandus** [comment]
You either be stuffed or hungry. Not both. Hunger is not determined by kcal, or at least it shouldn't. Malnutrition can happen even when you are stuffed full, especially when eating filling but not calories full food (plants). That is why the current system is wonky. You either fill your stomach or you don't. It is not depending on calories.

**Comment by Night-Pryanik** [comment]
Should be closed after #24210?

**Comment by Lamandus** [comment]
> 
> 
> Should be closed after #24210?

my "complain" is made after the merge... but yea, why not.
